src/crypto/Signature.ts

```diff
@@ -51,3 +117,60 @@ export default class Signature implements signature.Signature {
     return new Signature(signature.Signature.decode(bytes))
   }
 }
+
+// A signer that can be used to sign public keys.
+export interface KeySigner {
+  signKey(key: UnsignedPublicKey): Promise<SignedPublicKey>
+}
+
+// A wallet based KeySigner.
+export class WalletSigner implements KeySigner {
```

Comment:
This new class is a nice improvement. Always was a little awkward having the signature text be included in the key class.